docs/token-builder.md

```diff
@@ -0,0 +1,288 @@
+# Token Builder API
+
+## Overview
+
+The Token Builder is comprised of two parts: the `RequestBuilder` class and a variety of `Token` type classes.
+
+Each of the token classes has its own set of validations which are run when the token is instantiated. Any errors are added to an `errors` property on that instance of the token class.
+
+In `sorcerer`, the Token Parser will extract the tokens from request body and replace any occurrence of the token in an API data source's `url`, `body`, or `headers` with the token's computed value.
+
+## Token Types
+
+The Token Builder API includes several token utility classes, each serving a unique purpose.
+
+Currently supported tokens are:
+
+- ReplaceToken
+- ReplaceLargeToken
+- SecretToken
+- HmacToken
+- Sha1Token
+
+### ReplaceToken
+
+Replaces token with the `value` included in the token options. The length of this value must be less than 100 characters.
+
+Example:
+
+```jsx
+const options = {
+  name: 'FavoriteBand',
+  cacheOverride: 'Movable Band',
+  value: 'Beatles',
+  skipCache: true,
+};
+
+const tokenModel = new ReplaceToken(options);
+
+tokenModel.toJSON() // returns:
+// {
+//   name: 'FavoriteBand',
+//   type: 'replace',
+//   cacheOverride: 'Movable Band',
+//   skipCache: true,
+//   value: 'Beatles',
+// };
+```
+
+### ReplaceLargeToken
+
+Functionally, this behaves the same as the `ReplaceToken`, with the difference being that this token should only be used if the length of the replace value is greater than a specified character limit
+
+**Example:**
+
+```jsx
+const replaceValue = "some really long string"
+
+const options = {
+  name: 'FavoriteBand',
+  cacheOverride: 'Movable Band',
+  value: replaceValue,
+  skipCache: true,
+};
+
+const tokenModel = new ReplaceLargeToken(options);
+
+tokenModel.toJSON() // returns:
+// {
+//   name: 'FavoriteBand',
+//   type: 'replace',
```

Comment:
fix: this should result in `replaceLarge` or whichever is the one that `sorcerer` considers is "replace with large value"